MensaGuthaben/src/main/java/de/yazo_games/mensaguthaben/MainActivity.java

```diff
@@ -49,181 +49,199 @@
-        AutostartRegister.register(getPackageManager(),autostart);
+        }
 
-		Toolbar t = (Toolbar) findViewById(R.id.toolbar);
-		setSupportActionBar(t);
-		ViewCompat.setTransitionName(t,"toolbar");
+        Boolean autostart = PreferenceManager.getDefaultSharedPreferences(this).getBoolean("autostart", true);
+        AutostartRegister.register(getPackageManager(), autostart);
 
-		mAdapter = NfcAdapter.getDefaultAdapter(this);
+        mAdapter = NfcAdapter.getDefaultAdapter(this);
         mIntentFilter = new IntentFilter("android.nfc.action.ADAPTER_STATE_CHANGED");
 
 
-		// Create a generic PendingIntent that will be deliver to this activity.
-		// The NFC stack
-		// will fill in the intent with the details of the discovered tag before
-		// delivering to
-		// this activity.
-		mPendingIntent = PendingIntent.getActivity(this, 0, new Intent(this,
-				getClass()).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), 0);
+        // Create a generic PendingIntent that will be deliver to this activity.
+        // The NFC stack
+        // will fill in the intent with the details of the discovered tag before
+        // delivering to
+        // this activity.
+        mPendingIntent = PendingIntent.getActivity(this, 0, new Intent(this,
+                getClass()).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), PendingIntent.FLAG_IMMUTABLE);
 
 
-		// Setup an intent filter
-		IntentFilter tech = new IntentFilter(NfcAdapter.ACTION_TECH_DISCOVERED);
-		mFilters = new IntentFilter[] { tech, };
-		mTechLists = new String[][] { new String[] { IsoDep.class.getName(),
-				NfcA.class.getName() } };
+        // Setup an intent filter
+        IntentFilter tech = new IntentFilter(NfcAdapter.ACTION_TECH_DISCOVERED);
+        mFilters = new IntentFilter[]{tech,};
+        mTechLists = new String[][]{new String[]{IsoDep.class.getName(),
+                NfcA.class.getName()}};
 
-		if (getIntent().getAction().equals(ACTION_FULLSCREEN) &&!hasNewData) {
-			ValueData valueData = (ValueData) getIntent().getSerializableExtra(EXTRA_VALUE);
-			Log.w(TAG,"restoring data for fullscreen");
-			valueFragment.setValueData(valueData);
+        if (getIntent().getAction().equals(ACTION_FULLSCREEN) && !hasNewData) {
+            ValueData valueData = (ValueData) getIntent().getSerializableExtra(EXTRA_VALUE);
+            Log.w(TAG, "restoring data for fullscreen");
+            valueFragment.setValueData(valueData);
 
-		}
-	}
+        }
 
-	boolean hasNewData = false;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            mAdapter.enableReaderMode(this, this,
+                    NfcAdapter.FLAG_READER_NFC_A |
+                            NfcAdapter.FLAG_READER_NFC_B |
+                            NfcAdapter.FLAG_READER_NFC_F |
+                            NfcAdapter.FLAG_READER_NFC_V |
+                            NfcAdapter.FLAG_READER_NFC_BARCODE |
+                            NfcAdapter.FLAG_READER_NO_PLATFORM_SOUNDS, null);
```

Comment:
```suggestion
                            NfcAdapter.FLAG_READER_NFC_BARCODE, null);
```
You disable sound/vibration on tag scan here. Is this intended?

Reply:
Thanks, I completely missed that.